### PR TITLE
MINIFICPP-2106 Add 'Use Path Style Access' property to PutS3Object

### PR DESCRIPTION
**Pull request by lordgamez**
Tested on a live AWS instance, when the `Use Path Style Access` was set to true, the `s3.eu-west-2.amazonaws.com` URL was used and the bucket name was set in the path (path style access), otherwise the virtual path URL was used `my-bucket.s3.eu-west-2.amazonaws.com`

https://issues.apache.org/jira/browse/MINIFICPP-2106

-----------------------------------
Thank you for submitting a contribution to Apache NiFi - MiNiFi C++.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [ ] Is there a JIRA ticket associated with this PR? Is it referenced
     in the commit message?

- [ ] Does your PR title start with MINIFICPP-XXXX where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [ ] Has your PR been rebased against the latest commit within the target branch (typically main)?

- [ ] Is your initial contribution a single, squashed commit?

### For code changes:
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?
- [ ] If applicable, have you updated the LICENSE file?
- [ ] If applicable, have you updated the NOTICE file?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check GitHub Actions CI results for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

File: extensions/aws/tests/MockS3RequestSender.h [hunk 1]
```diff
@@ -250,6 +252,10 @@ class MockS3RequestSender : public minifi::aws::s3::S3RequestSender {
     return client_config_;
   }
 
+  bool getUserVirtualAddressing() const {
```
Comment (szaszm) [inline extensions/aws/tests/MockS3RequestSender.h:255]: This looks like a typo
```suggestion
  bool getUseVirtualAddressing() const {
```

Comment (lordgamez) [reply extensions/aws/tests/MockS3RequestSender.h]: Good catch, fixed in 70dd30bb57534a2eb38d3fc5cb76150900a8c076

**Review by martinzink — APPROVED** [review]
small nitpick but otherwise LGTM

File: extensions/aws/processors/PutS3Object.cpp [hunk 2]
```diff
@@ -73,6 +73,11 @@ void PutS3Object::onSchedule(const std::shared_ptr<core::ProcessContext> &contex
   }
   logger_->log_debug("PutS3Object: Server Side Encryption [%s]", server_side_encryption_);
 
+  bool use_path_style_access = false;
+  if (context->getProperty(UsePathStyleAccess.getName(), use_path_style_access)) {
+    use_virtual_addressing_ = !use_path_style_access;
+  }
+
```
[the remark this answers is not in the document]
Comment (lordgamez) [reply extensions/aws/processors/PutS3Object.cpp]: Updated in 0cc5232b2bda754eefbc48f7f9ff99fca40131f1

**Review by szaszm — APPROVED** [review]

**Pull request closed without merging**